cmd/frontrunning/main.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/ecdsa"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"log"
+	"math"
+	"math/big"
+	"net/http"
+	"time"
+
+	"github.com/Amenokal-Labs/mev-mantis.git/pkg/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/ethclient/gethclient"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+func main() {
+	ethclient, err := ethclient.Dial("https://goerli.infura.io/v3/" + utils.GetKey("INFURA_KEY"))
+	if err != nil {
+		log.Fatal("[0] ", err)
+
+	}
+
+	rpc, err := rpc.Dial("wss://goerli.infura.io/ws/v3/" + utils.GetKey("INFURA_KEY"))
+	if err != nil {
+		log.Fatal("[1] ", err)
+	}
+	subscriber := gethclient.New(rpc)
+	fmt.Println("client connected..")
+
+	// txs := make(chan *types.Transaction)
+	// _, err = rpcClient.SubscribeFullPendingTransactions(context.Background(), txs)
+	// if err != nil {
+	// 	panic(err)
+	// }
+
+	for {
+		pc, _ := ethclient.PendingTransactionCount(context.Background())
+		fmt.Println("\nPending count:", pc)
+
+		txn, marshalledTxn, from := getTxn(ethclient, subscriber)
+		calldata, contract := getCalldata(txn, marshalledTxn)
```

Comment:
could you add a comment below that explain why you're checking whether `contract == "0x"`?

Reply:
I assumed that if the contract has no code, then there's no logic to leak value to extract